source/common/upstream/upstream_impl.cc

```diff
@@ -67,6 +67,23 @@ HostImpl::createConnection(Event::Dispatcher& dispatcher, const ClusterInfo& clu
 
 void HostImpl::weight(uint32_t new_weight) { weight_ = std::max(1U, std::min(100U, new_weight)); }
 
+PrioritySetImpl::PrioritySetImpl() { getHostSet(0); }
+
+HostSet& PrioritySetImpl::getHostSet(uint32_t priority) {
+  if (host_sets_.size() < priority + 1) {
+    for (size_t i = host_sets_.size(); i <= priority; ++i) {
+      host_sets_.push_back(HostSetPtr{new HostSetImpl(i)});
+      host_sets_[i]->addMemberUpdateCb([this](uint32_t priority,
```

Comment:
This is another place it seems that `std::map` implementation might be cleaner, but not sure.

Reply:
I think map will be uglier when we start selecting priority levels, because you're not guaranteed in-order priorities or existence.  Vector with empty sets makes more sense to me.

Also one is not supposed to skip priorities, we just dont have config sanitization yet.

Reply:
FWIW, I don't think it's broken to skip priorities, in the same sense that it's not broken to assign processes to different priority levels in a scheduler where there is no contiguity. If it simplifies things I guess it's fine.

Reply:
proto validation will never do this type of checking for us. I would be fine having a check that all priorities need to be sequential and do this check in the EDS response. If we did this, we could also pre-create the host sets. I actually wonder if we really need the `getHostSet()` function which auto creates. It seems less error prone to just operate on the returned vector? Thoughts?

Reply:
I believe we can get rid of getHostSet from the API pretty easily, so that's done.

At this point we could replace getHostSet with
if (getHostSetsPerPriority.size() < priority) { createHostSet(priority)} but that seems suboptimal :-P
I'll rename getOrCreateHostSet to make the create explicit.  SG?